src/coinbase/address/developer_address.ts

```diff
@@ -0,0 +1,364 @@
+   * @param amount - The amount of the Asset to send.
+   * @param assetId - The ID of the Asset to send. For Ether, Coinbase.assets.Eth, Coinbase.assets.Gwei, and Coinbase.assets.Wei supported.
+   * @param destination - The destination of the transfer. If a Wallet, sends to the Wallet's default address. If a String, interprets it as the address ID.
+   * @param intervalSeconds - The interval at which to poll the Network for Transfer status, in seconds.
+   * @param timeoutSeconds - The maximum amount of time to wait for the Transfer to complete, in seconds.
+   * @returns The transfer object.
+   * @throws {APIError} if the API request to create a Transfer fails.
+   * @throws {APIError} if the API request to broadcast a Transfer fails.
+   * @throws {Error} if the Transfer times out.
+   */
+  public async createTransfer(
+    amount: Amount,
+    assetId: string,
+    destination: Destination,
+    intervalSeconds = 0.2,
+    timeoutSeconds = 10,
+  ): Promise<Transfer> {
+    if (!Coinbase.useServerSigner && !this._key) {
+      throw new InternalError("Cannot transfer from address without private key loaded");
+    }
+    const asset = await Asset.fetch(this.getNetworkId(), assetId);
+    const [destinationAddress, destinationNetworkId] =
+      this.getDestinationAddressAndNetwork(destination);
+
+    const normalizedAmount = new Decimal(amount.toString());
+    const currentBalance = await this.getBalance(assetId);
+    if (currentBalance.lessThan(normalizedAmount)) {
+      throw new ArgumentError(
+        `Insufficient funds: ${normalizedAmount} requested, but only ${currentBalance} available`,
+      );
+    }
+
+    const createTransferRequest = {
+      amount: asset.toAtomicAmount(normalizedAmount).toString(),
+      network_id: destinationNetworkId,
+      asset_id: asset.primaryDenomination(),
+      destination: destinationAddress,
+    };
+
+    let response = await Coinbase.apiClients.transfer!.createTransfer(
+      this.getWalletId(),
+      this.getId(),
+      createTransferRequest,
+    );
+
+    let transfer = Transfer.fromModel(response.data);
+    const wallet = new ethers.Wallet(this._key!.privateKey);
+
+    if (!Coinbase.useServerSigner) {
+      const transaction = transfer.getTransaction();
+      let signedPayload = await wallet!.signTransaction(transaction);
+      signedPayload = signedPayload.slice(2);
+
+      const broadcastTransferRequest = {
+        signed_payload: signedPayload,
+      };
+
+      response = await Coinbase.apiClients.transfer!.broadcastTransfer(
+        this.getWalletId(),
+        this.getId(),
+        transfer.getId(),
+        broadcastTransferRequest,
+      );
+
+      transfer = Transfer.fromModel(response.data);
+    }
+
+    const startTime = Date.now();
+    while (Date.now() - startTime < timeoutSeconds * 1000) {
+      await transfer.reload();
+      const status = transfer.getStatus();
+      if (status === TransferStatus.COMPLETE || status === TransferStatus.FAILED) {
```

Comment:
Do we want the entire `transfer` piece to enforce blocking like this?

Would we not want to potentially get the `transfer` object back quicker and then await on it completing?

That would allow for developers to at least have one process that could update a record on their side and then optionally choose to wait?

E.g. what if this takes 2 minutes to land on-chain on a slower network?

Reply:
Makes sense, I can have a follow up on this in a separate PR 👍